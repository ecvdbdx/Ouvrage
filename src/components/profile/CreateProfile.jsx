import React from 'react';
import Joi from 'joi';
import base from '../../airtable/config.js';

const schema = {
  firstName: Joi.string().required(),
  username: Joi.string().required(),
  lastName: Joi.string().required(),
  email: Joi.string().email().required(),
  birthdate: Joi.string().isoDate(),
  avatar: Joi.string()
};

export default class CreateProfile extends React.Component {
  constructor () {
    super();

    this.state = {
      firstName: '',
      username: '',
      lastName: '',
      email: '',
      birthdate: '',
      avatar: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.validate = this.validate.bind(this);
    this.submit = this.submit.bind(this);
  }

  validate () {
    return Joi.validate(this.state, schema);
  }

  handleChange (event, field) {
    this.setState({
      [field]: event.target.value
    });
  }

  submit () {
    this.validate()
      .then((value) => {
        console.log(value);

        base('Profil').create({
          firstName: value.firstName,
          lastName: value.lastName,
          username: value.username,
          email: value.email,
          birthdate: value.birthdate,
          avatar: value.avatar
        }, function (err) {
          if (err)
            console.error(err);
        });
      })
      .catch((error) => {
        console.error(error);
      });
  }

  render () {
    const { firstName, username, lastName, email, birthdate, avatar } = this.state;

    return (
      <div>
        <form>
          <input name="first_name" type="text" placeholder="Firstname"
            value={firstName}
            onChange={e => this.handleChange(e, 'firstName')}
          />
          <input name="username" type="text" placeholder="Username"
            value={username}
            onChange={e => this.handleChange(e, 'username')}
          />
          <input name="last_name" type="text" placeholder="Lastname"
            value={lastName}
            onChange={e => this.handleChange(e, 'lastName')}
          />
          <input name="email" type="email" placeholder="Email"
            value={email}
            onChange={e => this.handleChange(e, 'email')}
          />
          <input name="birthdate" type="date" placeholder="Birthdate"
            value={birthdate}
            onChange={e => this.handleChange(e, 'birthdate')}
          />
          <input name="avatar" type="file"
            value={avatar}
            onChange={e => this.handleChange(e, 'avatar')}
          />
        </form>

        <button onClick={this.submit} >Add profil</button>
      </div>
    );
  }
}
