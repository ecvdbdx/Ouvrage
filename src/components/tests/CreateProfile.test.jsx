import React from 'react';
import { shallow } from 'enzyme';

import CreateProfile from './../profile/CreateProfile.jsx';

it('renders correctly', () => {
  const wrapper = shallow(<CreateProfile />);
  expect(wrapper).toMatchSnapshot();
});
