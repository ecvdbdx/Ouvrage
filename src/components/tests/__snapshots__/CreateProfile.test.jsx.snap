// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateProfile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form>
          <input
            name="first_name"
            onChange={[Function]}
            placeholder="Firstname"
            type="text"
            value=""
          />
          <input
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
          <input
            name="last_name"
            onChange={[Function]}
            placeholder="Lastname"
            type="text"
            value=""
          />
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
          <input
            name="birthdate"
            onChange={[Function]}
            placeholder="Birthdate"
            type="date"
            value=""
          />
          <input
            name="avatar"
            onChange={[Function]}
            type="file"
            value=""
          />
        </form>,
        <button
          onClick={[Function]}
        >
          Add profil
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="first_name"
              onChange={[Function]}
              placeholder="Firstname"
              type="text"
              value=""
            />,
            <input
              name="username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />,
            <input
              name="last_name"
              onChange={[Function]}
              placeholder="Lastname"
              type="text"
              value=""
            />,
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            />,
            <input
              name="birthdate"
              onChange={[Function]}
              placeholder="Birthdate"
              type="date"
              value=""
            />,
            <input
              name="avatar"
              onChange={[Function]}
              type="file"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "first_name",
              "onChange": [Function],
              "placeholder": "Firstname",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "username",
              "onChange": [Function],
              "placeholder": "Username",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "last_name",
              "onChange": [Function],
              "placeholder": "Lastname",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "birthdate",
              "onChange": [Function],
              "placeholder": "Birthdate",
              "type": "date",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "avatar",
              "onChange": [Function],
              "type": "file",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add profil",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add profil",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form>
            <input
              name="first_name"
              onChange={[Function]}
              placeholder="Firstname"
              type="text"
              value=""
            />
            <input
              name="username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
            <input
              name="last_name"
              onChange={[Function]}
              placeholder="Lastname"
              type="text"
              value=""
            />
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            />
            <input
              name="birthdate"
              onChange={[Function]}
              placeholder="Birthdate"
              type="date"
              value=""
            />
            <input
              name="avatar"
              onChange={[Function]}
              type="file"
              value=""
            />
          </form>,
          <button
            onClick={[Function]}
          >
            Add profil
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="first_name"
                onChange={[Function]}
                placeholder="Firstname"
                type="text"
                value=""
              />,
              <input
                name="username"
                onChange={[Function]}
                placeholder="Username"
                type="text"
                value=""
              />,
              <input
                name="last_name"
                onChange={[Function]}
                placeholder="Lastname"
                type="text"
                value=""
              />,
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
              />,
              <input
                name="birthdate"
                onChange={[Function]}
                placeholder="Birthdate"
                type="date"
                value=""
              />,
              <input
                name="avatar"
                onChange={[Function]}
                type="file"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "first_name",
                "onChange": [Function],
                "placeholder": "Firstname",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "username",
                "onChange": [Function],
                "placeholder": "Username",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "last_name",
                "onChange": [Function],
                "placeholder": "Lastname",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "birthdate",
                "onChange": [Function],
                "placeholder": "Birthdate",
                "type": "date",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "avatar",
                "onChange": [Function],
                "type": "file",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add profil",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add profil",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
